### PR TITLE
[Feat][Bugfix] Enable additional dimension for Flashinfer MLA and fix routing dtype

### PR DESCRIPTION
**Pull request by dbari**
## Purpose

This PR implements the following:
- Enable additional `qk_nope_head_dim=64` for Flashinfer MLA, supported starting 0.6.6
- Fix routing logits dtype in Flashinfer MoE for models not using `DeepseekV3` routing, e.g. Mistral Large 3 that uses `Renormalize`
- Fix a minor bug in torch compilation, when both `input_ids` and `inputs_embeds` are `None` in `DeepseekV2Model.forward()`

## Test Plan

Test bugfix by running inference on `mistralai/Mistral-Large-3-675B-Instruct-2512-NVFP4`.

## Test Result

Current nightly `vllm/vllm-openai:nightly-36735fd77224467e6580f3bd48eb32d4fca8c72e` returns an empty string:
```
$ vllm chat -q "What is the capital of France?"

$
```

Image built from this PR:
```
$ vllm chat -q "What is the capital of France?"
The capital of France is **Paris**.
$
```

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/models/deepseek_v2.py [hunk 1]
```diff
@@ -333,8 +337,12 @@ def __init__(
         # NOTE(rob): this is a hack until we finish off the PR for
         # merging TRTLLM kernels into the MK framework. Then we can
         # query the MonolithicMK for the expected router logits.
+        # NOTE(dbari): Use BF16 if routing is not Deepseek, e.g. Mistral Large 3
```
Comment (pavanimajety) [inline vllm/model_executor/models/deepseek_v2.py:340]: @robertgshaw2-redhat  for viz & review

File: vllm/model_executor/models/deepseek_v2.py [hunk 2]
```diff
@@ -1197,6 +1205,11 @@ def forward(
             if inputs_embeds is not None:
                 hidden_states = inputs_embeds
             else:
+                if input_ids is None:
+                    raise ValueError(
+                        "Either input_ids or inputs_embeds must be provided "
+                        "to DeepseekV2Model.forward"
+                    )
```
Comment (pavanimajety) [inline vllm/model_executor/models/deepseek_v2.py:1212]: when would we run into this?

File: vllm/model_executor/models/deepseek_v2.py [hunk 3]
```diff
@@ -333,8 +337,12 @@ def __init__(
         # NOTE(rob): this is a hack until we finish off the PR for
         # merging TRTLLM kernels into the MK framework. Then we can
         # query the MonolithicMK for the expected router logits.
+        # NOTE(dbari): Use BF16 if routing is not Deepseek, e.g. Mistral Large 3
         self.gate.set_out_dtype(
-            torch.float32 if self.experts.quant_method.is_monolithic else torch.bfloat16
+            torch.float32
+            if self.experts.quant_method.is_monolithic
+            and self.experts.routing_method_type == RoutingMethodType.DeepSeekV3
+            else torch.bfloat16
```
Comment (pavanimajety) [inline vllm/model_executor/models/deepseek_v2.py:345]: @robertgshaw2-redhat shouldn't `torch.float32` be default for all Deepseek Architectures (i.e. when RoutingMethodType==DeepseekV3) and not just flashinfer trtllm / monolithic kernels?

**Review by pavanimajety — APPROVED** [review]
LGTM, @dbari, Thanks for the PR.

File: vllm/model_executor/models/deepseek_v2.py (hunk 3 above)
Comment (robertgshaw2-redhat) [reply vllm/model_executor/models/deepseek_v2.py:345]: I don’t think the grouped took kernels in vLLM support fp32. Ideally we could add this support. Need to double checj

File: vllm/model_executor/models/deepseek_v2.py (hunk 2 above)
Comment (dbari) [reply vllm/model_executor/models/deepseek_v2.py:1212]: I ran into an error of the compilation, where the function was called with both arguments `None` and the compilation failed due to a code path trying to call `input_ids.shape` deeper in the code. Raising a `ValueError` is accepted without errors by the compiler, the `AttributeError` from `None.shape` does not.

Unfortunately I don't have a reproducer at hand.
